config/initializers/errors.rb

```diff
@@ -12,5 +12,37 @@ def initialize(message="Invalid SQL query for this Live Query")
         super
       end
     end
+
+    class ExpectedError < StandardError
+      attr_reader :error_code, :status, :message
+
+      def initialize(error_code, status, message)
+        @error_code = error_code
+        @status = status
+        @message = message
+      end
+
+      def display_error()
+        recursively_print(self)
+      end
+
+      def recursively_print(error, margin="")
```

Comment:
It will display recursively all the resons and the the stack trace of the deeper exception (since the higher stacks are contained in the deepest).

Example:
```
Cannot reach the forest server
which was caused by:
 Unauthorized
...then all stacktrace...
```